CONTRIBUTING.md

```diff
@@ -0,0 +1,76 @@
+# Contributing to Twinkle
+
+:tada::tada: Thanks for taking the time to contribute! :tada::tada:
+
+There are many ways to help out!
+
+## Bug reports and feature requests
+
+If you think you've found a bug or have a great idea for a new feature, great!  You can [open a new GitHub issue here](https://github.com/azatoth/twinkle/issues/new) (GitHub account required) or report it at [Wikipedia talk:Twinkle][].  Bigger changes or more complicated requests should be made on-wiki so other users can take part in the discussion of your feature proposal.  If you're unsure if something is a bug, other editors may be able to help identify the issue.  Be sure to search the talk page archives and GitHub issues to see if your request has already been discussed in the past.
+
+Whatever the case, the more detailed your description the easier it will be to respond to your report or request.
+
+### Reporting a bug
+
+A good bug report will include:
+- A brief, descriptive title that mentions the module you were using (tag, CSD, xfd, etc.).
+- The steps leading up to the issue so we can replicate it.  This should include the page and revision you were on, the action you were performing, and the options you selected.
+- Any errors or messages that Twinkle reported an error when it got stuck.
+- What you think *should* have happened.
+- Anything you can find in your [browser's console window][jserrors].
+
+## Contributing a pull request
+### Getting started
+
+If you'd like to help with Twinkle's development, wonderful!  Anyone can contribute, and it's easy to get set up to do so.
+
+First, familiarize yourself with the code; most likely, the changes you want are to one of the [modules](./modules); you can also check out the [individual Gadget pages][twinkle_gadget] onwiki.  If you want to propose changes yourself, [fork the repository](https://help.github.com/articles/fork-a-repo/) to make sure you always have the latest versions.  If you're new to GitHub or Git in general, you probably want to read [Getting started with GitHub](https://help.github.com/en/github/getting-started-with-github) first.
+
+Once you've got a local fork up and running, commit your changes!
+
+
+### Testing your code
+
+Testing Twinkle can be tricky, but the best way to test your code is to open up your [browser's console window][jserrors] and paste in your new code.  You'll have to load the new version by running the corresponding function in your console, e.g., `Twinkle.protect()` for twinkleprotect.js.
```

Comment:
YMMV, although I agree.  Going with "straightforward" as I try to avoid using the word "simple" in instructional material.

I *do* plan to incorporate some of the excellent material you added to the wiki, but was leaving that for a later update to DEVELOPER.md with some other things I want to document.

```suggestion
Testing Twinkle can be tricky, but the most straightforward way to test your code is to open up your [browser's console window][jserrors] and paste in your new code.  You'll have to load the new version by running the corresponding function in your console, e.g., `Twinkle.protect()` for twinkleprotect.js.
```

Reply:
Importing the code from your local directly into the production wiki is a clever idea! Some years ago I asked others how they did testing/feature development, and they were all using the console. Will this still work after the [CSP header stuff](https://phabricator.wikimedia.org/T28508) goes live? At any rate, I agree it's worth mentioning here, now or in a separate PR.

Reply:
I left a note at [phab](https://phabricator.wikimedia.org/T135963#5610199) requesting localhost from being allowed in CSP. APerson [did the same](https://phabricator.wikimedia.org/T135963#4843260) some time back it seems to have been ignored.